snuba/clickhouse/formatter/query.py

```diff
@@ -124,24 +138,32 @@ def _format_query_content(
 
 
 def _format_select(
-    query: AbstractQuery, formatter: ExpressionVisitor[str]
+    query: AbstractQuery, formatter: ExpressionVisitor[str], sort_fields: bool = False
 ) -> StringNode:
     selected_cols = [
         e.expression.accept(formatter) for e in query.get_selected_columns()
     ]
+    if sort_fields:
+        selected_cols = sorted(selected_cols)
     return StringNode(f"SELECT {', '.join(selected_cols)}")
 
 
 def _build_optional_string_node(
     name: str,
     expression: Optional[Expression],
     formatter: ExpressionVisitor[str],
+    sort_fields: bool = False,
 ) -> Optional[StringNode]:
-    return (
-        StringNode(f"{name} {expression.accept(formatter)}")
-        if expression is not None
-        else None
-    )
+    if expression is not None:
+        expression_string = expression.accept(formatter)
+        # Sorting will not work if sub-queries in expressions becomes supported in the future
+        if sort_fields:
+            expression_list = re.split("(?=AND)|(?=OR)", expression_string)
```

Comment:
FYI, best practice is to compile regexes globally, then use the compiled version here.